modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -755,8 +752,7 @@ public static void setHeadersToTransportMessage(HTTPCarbonMessage cMsg, BStruct
             return null;
         }
         HttpHeaders removedHeaders = new DefaultHttpHeaders();
-        if (struct.getNativeData(Constants.OUTBOUND_RESPONSE) == null
-                && !struct.getStringField(Constants.RESPONSE_SERVER_INDEX).equals("")) {
+        if (!struct.getStringField(Constants.RESPONSE_SERVER_INDEX).equals("")) {
```

Comment:
Can getStringField return null here? Also isEmpty() would be better

Reply:
getStringField is initialized as an empty string so it won't be a null. I'll change equals() to isEmpty()